### PR TITLE
Switch to AIDs form HIDs

### PR DESCRIPTION
**Pull request by MuncleUscles**
Switch to AIDs

### CODE REVIEW

File: src/modules/atommodel/RadixAtomModel.spec.ts [hunk 1]
```diff
@@ -1,16 +1,20 @@
...
 
     
     {
         const atom = new RadixAtom()
+        atom.particleGroups = [new RadixParticleGroup([RadixSpunParticle.up(
+            new RadixUniqueParticle(RadixAddress.generateNew(), 'test'),
+        )])]
         
         it(`should compute hid`, () => {
-            expect(atom.getHID.bind(atom)).to.not.throw()
```
Comment (Sajjon) [inline src/modules/atommodel/RadixAtomModel.spec.ts]: Can we maybe test more than just `not throw`, maybe at least that the text is not empty? Or better yet that it is a hex string of correct length?

File: src/modules/universe/RadixNode.ts [hunk 2]
```diff
@@ -2,21 +2,25 @@ import { RadixNodeInfo } from '../..';
...
+        const remainder = shard.mod(RadixNode.SHARD_CHUNK_HALF_RANGE)
+
         if (this.nodeInfo.system) {
-            const low = Long.fromValue(this.nodeInfo.system.shards.low)
-            const high = Long.fromValue(this.nodeInfo.system.shards.high)
+            const low = Long.fromValue(this.nodeInfo.system.shards.range.low)
+            const high = Long.fromValue(this.nodeInfo.system.shards.range.high)
 
             if (high.lessThan(low)) {
                 // Wrap around
```
Comment (Sajjon) [inline src/modules/universe/RadixNode.ts]: I'm curious, I have not seen Java lib wrapping around the range, is this documented somewhere (WP, Confluence?), maybe I've just miseed that Java wraps...?

File: src/modules/universe/RadixNode.ts [hunk 3]
```diff
@@ -2,21 +2,25 @@ import { RadixNodeInfo } from '../..';
 import Long from 'long';
 
 export default class RadixNode {
+    public static SHARD_CHUNK_HALF_RANGE = Long.fromValue(1).shl(43)
+
     constructor(readonly nodeInfo: RadixNodeInfo, readonly wsAddress: string, readonly httpAddress: string) {}
 
     public canServiceShard(shard: Long): boolean {
```
Comment (Sajjon) [inline src/modules/universe/RadixNode.ts]: Is there any documented pseudocode for `canServiceShard` function somehere? :) Maybe in WP, Confluence? I don't think I've seen it.

File: src/modules/atommodel/primitives/RadixAID.ts [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+
+
+    public static from(hash: Buffer, shards: long[]) {
+        const shardIndex = hash[0] % shards.length
+
+        const selectedShard = shards
+            .map(s => s.toUnsigned())
+            .sort((a, b) => a.compare(b))
+            .map(s => s.toSigned())
+            [shardIndex]
```
[the remark this answers is not in the document]
Comment (MuncleUscles) [reply src/modules/atommodel/primitives/RadixAID.ts]: Good point, there is no protection

File: src/modules/txbuilder/RadixTransactionBuilder.ts [hunk 5]
```diff
@@ -553,7 +553,28 @@ export default class RadixTransactionBuilder {
...
 
-    public addMessageParticle(from: RadixAccount, data: string | Buffer, metadata: any, recipients: RadixAccount[]) {
+
+    public addUnencryptedMessage(
+        from: RadixAccount,
+        applicationId: string,
+        message: string,
+        recipients: RadixAccount[],
+    ) {
+        const recipientPubKeys = recipients.map(r => r.address.getPublic())
```
[the remark this answers is not in the document]
Comment (MuncleUscles) [reply src/modules/txbuilder/RadixTransactionBuilder.ts]: Whoops, yep

File: src/modules/universe/RadixNode.ts (hunk 3 above)
Comment (MuncleUscles) [reply src/modules/universe/RadixNode.ts]: It's the same as `ShardSpace::intersects` in java, I have not generalised it

File: src/modules/universe/RadixNode.ts (hunk 2 above)
Comment (MuncleUscles) [reply src/modules/universe/RadixNode.ts]: No, Java doesn't wrap, I coded this before the current implementation. The core enforces `low<high`, not sure if we're going to get back to wraparound, I think we should though

Comment (Sajjon) [reply src/modules/universe/RadixNode.ts]: I also enforce low<high, we might wanna add a ticket about this so we consolidate the three libraries (possible forth: Kotlin) :)

File: src/modules/universe/RadixNode.ts (hunk 3 above)
Comment (Sajjon) [reply src/modules/universe/RadixNode.ts]: Ah thx!

File: src/modules/atommodel/RadixAtomModel.spec.ts (hunk 1 above)
Comment (MuncleUscles) [reply src/modules/atommodel/RadixAtomModel.spec.ts]: Might as well, this was initially introduced here and in all the particles for verifying the serialization pipeline mostly, but it's as good a spot as any for checking some more stuff about AIDs.

File: src/modules/universe/RadixNode.ts (hunk 2 above)
Comment (MuncleUscles) [reply src/modules/universe/RadixNode.ts]: Would be good to get a document describing sharding in detail, how it works now and how it will work by launch

Comment (Sajjon) [reply src/modules/universe/RadixNode.ts]: agreed!

**Pull request merged**